libs/util/shared/src/lib/services/proxy.service.ts

```diff
@@ -15,10 +15,15 @@ export class ProxyService {
    */
   getProxiedUrl(url: string): string {
     if (!this.proxyPath) return url
-    const proxyUrl = new URL(
-      this.proxyPath,
-      window.location.toString()
-    ).toString()
+    const urlObj = new URL(url)
+    const current = window.location
+    const proxyUrl = new URL(this.proxyPath, current.toString()).toString()
+    if (
+      current.hostname === urlObj.hostname &&
+      current.protocol === urlObj.protocol
```

Comment:
We should also check that the port is the same, otherwise CORS will apply as well. E.g. going from http://localhost:8080 to http://localhost:8081 is considered a cross-origin request.